assets/js/googlesitekit/widgets/util/combine-widgets.test.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * Widgets combination utilities tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { combineWidgets } from './combine-widgets';
+import { getWidgetLayout } from './get-widget-layout';
+import { WIDTH_GRID_CLASS_MAP } from './constants';
+import { WIDGET_WIDTHS } from '../datastore/constants';
+import ReportZero from '../../../components/ReportZero';
+import ActivateModuleCTA from '../../../components/ActivateModuleCTA';
+import CompleteModuleActivationCTA from '../../../components/CompleteModuleActivationCTA';
+
+describe( 'combineWidgets', () => {
+	const getQuarterWidget = ( slug ) => ( { slug, width: WIDGET_WIDTHS.QUARTER } );
+	const getHalfWidget = ( slug ) => ( { slug, width: WIDGET_WIDTHS.HALF } );
+
+	const getRegularState = () => null;
+	const getReportZeroState = ( moduleSlug ) => ( { Component: ReportZero, metadata: { moduleSlug } } );
+	const getActivateModuleCTAState = ( moduleSlug ) => ( { Component: ActivateModuleCTA, metadata: { moduleSlug } } );
+	const getCompleteModuleActivationCTAState = ( moduleSlug ) => ( { Component: CompleteModuleActivationCTA, metadata: { moduleSlug } } );
+
+	// Every test case below corresponds to a matching story in `stories/widgets.stories.js` under
+	// "Global/Widgets/Widget Area/Special combination states".
```

Comment:
I was more thinking that coupling them will ensure we keep the coverage consistent. We don't need to test more scenarios of similar order etc., so I don't think using random other combinations would help. The comments here IMO are a reminder that we should keep the scenarios aligned. If there is a new relevant scenario to be tested, it should be added to both the Jest test and the Storybook story.

Reply:
I suppose that's okay in this case since the logic is very much visually-oriented. I don't think it's something that we want to maintain as a general convention however.